src/metabase/types.clj

```diff
@@ -5,6 +5,7 @@
 (derive :type/Dictionary :type/Collection)
 (derive :type/Array :type/Collection)
 
+(derive :type/Nil :type/*)
```

Comment:
why can't you just change the `fingerprinter` dispatch function to return `:type/*` instead of `nil` when it sees it for the special type? Then you wouldn't need to make any changes to any of the implementations